chromium/browser_type.go

```diff
@@ -356,24 +360,44 @@ func setFlagsFromArgs(flags map[string]any, args []string) {
 
 // setFlagsFromK6Options adds additional data to flags considering the k6 options.
 // Such as: "host-resolver-rules" for blocking requests.
-func setFlagsFromK6Options(flags map[string]any, k6opts *k6lib.Options) {
+func setFlagsFromK6Options(flags map[string]any, k6opts *k6lib.Options) error {
 	if k6opts == nil {
-		return
+		return nil
 	}
 
 	hostResolver := []string{}
 	if currHostResolver, ok := flags["host-resolver-rules"]; ok {
 		hostResolver = append(hostResolver, fmt.Sprintf("%s", currHostResolver))
 	}
 
-	for k, v := range k6opts.Hosts {
+	// Add the host resolver rules.
+	//
+	// This is done by marshaling the k6 hosts option to JSON and then
+	// unmarshaling it to a map[string]string. This is done because the
+	// k6 v0.42 changed Hosts from a map to types.NullHosts and doesn't
+	// expose the map anymore.
+	//
+	// TODO: A better way to do this would be to handle the resolver
+	// rules by communicating with Chromium (and then using Hosts's
+	// Match method) instead of passing the rules via the command line
+	// to Chromium.
```

Comment:
How would we communicate with the chromium? CDP? I don't know why that would be better than passing the rules via the command line. Maybe we need another issue for that? 🙂

Reply:
Right. I'll create another issue next week. Reminder set 👍😄